### PR TITLE
Fix ignoring changes to Query CircuitBreaker settings.

### PR DESCRIPTION
**Pull request by matriv**
@seut @mfussenegger @andreidan I wrote an IntegTest for the query breaker but I cannot find any way to test the change for the "http-result" in `SqlHttpHandlerTest`. Any suggestions for that?

### CODE REVIEW

**Comment by matriv** [comment]
@andreidan Thx, will fix!

File: sql/src/main/java/io/crate/execution/jobs/JobSetup.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ public JobSetup(Settings settings,
                     MapSideDataCollectOperation collectOperation,
                     ClusterService clusterService,
                     NodeJobsCounter nodeJobsCounter,
-                    CrateCircuitBreakerService breakerService,
+                    CrateCircuitBreakerService circuitBreakerService,
```
Comment (seut) [inline sql/src/main/java/io/crate/execution/jobs/JobSetup.java]: Using a `Function<String, CircuitBreaker>` here would also perfectly fit imho (decoupling, eliminates test mocking).

File: sql/src/main/java/io/crate/rest/action/SqlHttpHandler.java [hunk 2]
```diff
@@ -88,21 +90,21 @@
...
+    private final CircuitBreakerService circuitBreakerService;
     private final UserLookup userLookup;
     private final Netty4CorsConfig corsConfig;
 
     private Session session;
 
     SqlHttpHandler(Settings settings,
                    SQLOperations sqlOperations,
-                   CircuitBreaker circuitBreaker,
+                   CircuitBreakerService circuitBreakerService,
```
Comment (seut) [inline sql/src/main/java/io/crate/rest/action/SqlHttpHandler.java]: also here maybe a function makes sense

File: sql/src/test/java/io/crate/integrationtests/CircuitBreakerIntegrationTest.java [hunk 3]
```diff
@@ -62,4 +66,18 @@ public void testQueryBreakerIsDecrementedWhenQueryCompletes() {
...
         assertThat(queryBreaker.getUsed(), is(breakerBytesUsedBeforeQuery));
     }
+
+    @Test
+    public void testQueryBreakerIsUpdatedWhenSettingIsChanged() {
+        execute("create table t1 (text string) clustered into 1 shards");
+        ensureYellow();
+        execute("insert into t1 values ('this is some text'), ('other text')");
+        refresh();
+
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/CircuitBreakerIntegrationTest.java]: maybe we should validate that the initial setting is not `100b` by issuing the same `select ...` statement before adjusting the setting to really ensure that then change worked? (could be that now or then something changes and the initial setting could be `100b` by coincidence...

File: sql/src/main/java/io/crate/execution/jobs/JobSetup.java (hunk 1 above)
Comment (matriv) [reply sql/src/main/java/io/crate/execution/jobs/JobSetup.java]: Here we cannot use the Function as this kind of injection is not supported by Guice.

File: sql/src/main/java/io/crate/rest/action/SqlHttpHandler.java (hunk 2 above)
Comment (matriv) [reply sql/src/main/java/io/crate/rest/action/SqlHttpHandler.java]: Here yes!!

File: sql/src/main/java/io/crate/execution/jobs/JobSetup.java (hunk 1 above)
Comment (seut) [reply sql/src/main/java/io/crate/execution/jobs/JobSetup.java]: ah sry sure, didn't see that this is injected

**Comment by matriv** [comment]
@seut Thx for the suggestions. Fixup pushed!

**Comment by matriv** [comment]
Backported to `3.0` with https://github.com/crate/crate/commit/0b2bb155c7d83a9b4471c0c4d8a2caf70892cb71.
Backported to `2.3` with https://github.com/crate/crate/commit/0419229cb6e547b005b4d7f599e34aaf6412cf68.

**Pull request merged**